### PR TITLE
Upload logfile in github actions when integation failed and remove allow-failure jobs

### PR DESCRIPTION
**Pull request by liya2017**
This PR contains 2 tips:

1.Upload integration files in  github actions and remove usless variables from UnitTest job
2. Remove jobs `Cyclic_dev_dependencies` & `Latest_Linters` from CI

### CODE REVIEW

**Comment by nervos-bot-user** [comment]
Can one of the admins verify this patch?

**Comment by nervos-bot-user** [comment]
Can one of the admins verify this patch?

File: test/run.sh [hunk 1]
```diff
@@ -59,6 +60,22 @@ expect "sftp>"
...
+    cd -
+  fi
+# upload github actions log if test failed
+  if [ -n "${BUILD_BUILDID:-}" ] && [ -n "${LOGBAK_SERVER:-}" ]; then
+    upload_id="github-actions-${test_id}-${BUILD_BUILDID:-0}-${ImageOS:-unknown}"
+    cd "${test_tmp_dir}"/..
+    tar -czf "${upload_id}.tgz" "${test_id}"
+    expect <<EOF
+spawn sftp -o "StrictHostKeyChecking=no" "${LOGBAK_USER}@${LOGBAK_SERVER}"
+expect "assword:"
```
Comment (quake) [inline test/run.sh:73]:
```suggestion
expect "password:"
```

Comment (yangby-cryptape) [reply test/run.sh:73]: `expect "assword:"` is better.

Some implementations are `Password:` and others are `password:`.

Comment (quake) [reply test/run.sh:73]: Got a tip

**Review by quake — APPROVED** [review]

**Review by zhangsoledad — APPROVED** [review]

**Comment by quake** [comment]
bors r=quake,zhangsoledad

**Review by doitian — APPROVED** [review]

**Pull request merged**